src/db.rs

```diff
@@ -71,6 +71,59 @@ impl DB {
         Ok(())
     }
 
+    pub fn migrate<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        if !self.dirs.is_empty() {
+            return Err(anyhow!(
+                "To prevent conflicts, you can only migrate from z with an empty \
+                 zoxide database!"
+            ));
+        }
+
+        let zdata = File::open(path)?;
+        let reader = BufReader::new(zdata);
+
+        for line in reader.lines().filter_map(|l| l.ok()) {
+            let first_pipe_idx = line
```

Comment:
Rather than working with indices, how about the more concise:

```rust
for line_res in rd.lines() {
    let line = line_res.with_context(|| anyhow!("could not read line"))?;
    let line_split = line.rsplitn(3, '|').collect::<Vec<&str>>();

    match line_split.as_slice() {
        [time_str, rank_str, path_str] => todo!(),
        [] | [""] => (),
        _ => return Err(todo!()),
    };
}
```

Reply:
Any reason why you chose `rsplitn` over `splitn`?

Reply:
The path name might have a `|` in it, but the timestamp and rank won't. Using `rsplitn` will take the `timestamp` and `rank` out first, and then use the rest of the line as the `path`.